async_simple/util/ThreadPool.h

```diff
@@ -93,19 +115,41 @@ inline ThreadPool::ThreadPool(size_t threadNum, bool enableWorkSteal)
 
             // If _enableWorkSteal false or work steal failed, wait for a pop
             // task.
-            if (!workerItem.fn && !_queues[id].pop(workerItem)) {
-                break;
-            }
+            if (!workerItem.fn && !_queues[id].pop(workerItem))
+                continue;
 
-            if (workerItem.fn) {
+            if (workerItem.fn)
                 workerItem.fn();
-            }
         }
     };
 
     _threads.reserve(_threadNum);
+
+#ifdef __linux__
+    // Since the CPU IDs might not start at 0 and might not be continuous
+    // in the containers,
+    // we need to get the avaialable cpus at first.
+    std::vector<uint32_t> cpu_ids;
+    if (_enableCoreBindings)
+        getCurrentCpus(cpu_ids);
+#endif
+
     for (auto i = 0; i < _threadNum; ++i) {
         _threads.emplace_back(worker, i);
+
+#ifdef __linux__
+        if (!_enableCoreBindings)
+            continue;
+
+        // Run threads per core.
+        cpu_set_t cpuset;
+        CPU_ZERO(&cpuset);
+        CPU_SET(cpu_ids[i % cpu_ids.size()], &cpuset);
+        int rc = pthread_setaffinity_np(_threads[i].native_handle(),
```

Comment:
we can use  sched_setaffinity syscall to avoid include pthread，because already be linux system